src/firefly/java/edu/caltech/ipac/visualize/plot/FitsRead.java

```diff
@@ -988,30 +927,34 @@ public synchronized void doStretch(RangeValues rangeValues,
                                        int startPixel,
                                        int lastPixel,
                                        int startLine,
-                                       int lastLine, ImageMask[] lsstMasks){
-
-
-
-
-        Histogram hist= getHistogram();
-
-
-        double slow = getSlow(rangeValues, float1d, imageHeader, hist);
-        double shigh = getShigh(rangeValues, float1d, imageHeader, hist);
-        if (SUTDebug.isDebug()) {
-            printInfo(slow, shigh, imageHeader.bitpix, rangeValues);
-        }
+                                       int lastLine, ImageMask[] lsstMasks)  {
 
         byte blank_pixel_value = mapBlankToZero ? 0 : (byte) 255;
 
         int[] pixelhist = new int[256];
 
+        //covert the row mask to real mask
+        float[] fMasks = getDataFloat();
+        ArrayList <Float>  noneZerorData= new ArrayList<Float>();
+        for (int i=0; i<fMasks.length; i++){
+            if (fMasks[i]!=0){
+                noneZerorData.add(fMasks[i]) ;
+            }
+        }
 
+        float[] fdata = new float[noneZerorData.size()];
+        for (int i=0; i<noneZerorData.size(); i++){
+            fdata[i]=noneZerorData.get(i).floatValue();
+        }
 
 
-        stretchPixels(startPixel, lastPixel, startLine, lastLine, imageHeader.naxis1,
-                blank_pixel_value, float1d, masks, pixelData, pixelhist,  lsstMasks);
+        short[] masks= (short[]) ArrayFuncs.convertArray(float1d, Short.TYPE, true);
+        for (int i=0; i<fMasks.length; i++){
+            masks[i] = (short) fMasks[i];
+        }
```

Comment:
Sorry, this was part of the test codes which I compared the results.  Then I deleted only the comparison part and left this part accidentally.  I need to remove all the codes added when I debug.